webrender/res/brush_blend.glsl

```diff
@@ -148,10 +148,6 @@ vec3 LinearToSrgb(vec3 color) {
 Fragment brush_fs() {
     vec4 Cs = texture(sColor0, vUv);
 
-    if (Cs.a == 0.0) {
-        return Fragment(vec4(0.0)); // could also `discard`
-    }
-
     // Un-premultiply the input.
     float alpha = Cs.a;
     vec3 color = Cs.rgb / Cs.a;
```

Comment:
There's nothing preventing division by zero here right?

Reply:
yeah, looks like a problem

Reply:
I think that we need to compute the result (even with the division by zero) and select the result at the end depending on whether the alpha was zero using `mix` or some other trick that dxc is unlikely to rewrite into an early return or early branch (and check that dxc indeed doesn't introduce a branch).
If division by zero is a real issue even when masked out, we can always temporarily set the alpha to 1 and still replace the color at the end of the shader as long as we don't introduce a shortcut.